src/pages/workspace/categories/WorkspaceCategoriesPage.tsx

```diff
@@ -0,0 +1,122 @@
+import type {CentralPaneNavigatorParamList} from '@navigation/types';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type SCREENS from '@src/SCREENS';
+import type * as OnyxTypes from '@src/types/onyx';
+import withPolicyAccessOrNotFound from './withPolicyAccessOrNotFound';
+import type {WithWorkspaceAccessProps} from './withPolicyAccessOrNotFound';
+
+type PolicyForList = {
+    value: string;
+    text: string;
+    keyForList: string;
+    isSelected: boolean;
+    rightElement: React.ReactNode;
+};
+
+type WorkspaceCategoriesOnyxProps = {
+    /** Collection of categories attached to a policy */
+    policyCategories: OnyxEntry<OnyxTypes.PolicyCategories>;
+};
+
+type WorkspaceCategoriesPageProps = WorkspaceCategoriesOnyxProps & WithWorkspaceAccessProps & StackScreenProps<CentralPaneNavigatorParamList, typeof SCREENS.WORKSPACE.CATEGORIES>;
+
+function WorkspaceCategoriesPage({policyCategories}: WorkspaceCategoriesPageProps) {
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const {translate} = useLocalize();
+    const [selectedCategories, setSelectedCategories] = useState<string[]>([]);
+
+    const categoryList = useMemo<PolicyForList[]>(
+        () =>
+            Object.values(policyCategories ?? {}).map((value) => ({
+                value: value.name,
+                text: value.name,
+                keyForList: value.name,
+                isSelected: selectedCategories.includes(value.name),
+                rightElement: (
+                    <View style={styles.flexRow}>
+                        <Text style={[styles.disabledText, styles.alignSelfCenter]}>{value.enabled ? translate('workspace.common.enabled') : translate('workspace.common.disabled')}</Text>
+                        <View style={styles.p1}>
+                            <Icon
+                                src={Expensicons.ArrowRight}
+                                fill={theme.icon}
+                            />
+                        </View>
+                    </View>
+                ),
+            })),
+        [policyCategories, selectedCategories, styles.alignSelfCenter, styles.disabledText, styles.flexRow, styles.p1, theme.icon, translate],
+    );
+
+    const toggleCategory = (category: PolicyForList) => {
+        setSelectedCategories((prev) => {
+            if (prev.includes(category.value)) {
+                return prev.filter((item) => item !== category.value);
+            }
+            return [...prev, category.value];
+        });
+    };
+
+    const toggleAllCategories = () => {
+        const isAllSelected = categoryList.every((category) => category.isSelected);
+        if (isAllSelected) {
+            setSelectedCategories([]);
+        } else {
+            setSelectedCategories(categoryList.map((item) => item.value));
+        }
+    };
+
+    if (Object.keys(policyCategories ?? {}).length === 0) {
+        return <FullScreenLoadingIndicator />;
```

Comment:
If a policy doesn't have any categories, showing a perpetual loader is not a good UX imo.
Just want to confirm if this will be replaced later on

Reply:
An empty table with no rows.

The user can then add categories. cc: @Expensify/design

Reply:
Ah yeah, good point. It's somewhat unlikely to run into this, but totally possible.

- a workspace is created with a default set of categories 
- if somebody manually deletes all of the individual category values, the page will be empty but the categories feature still `enabled` until they disable the feature overall

So I think we should use the empty state for that: 

![image](https://github.com/Expensify/App/assets/16232057/8951f0de-f104-4ebd-a0de-5abcd9c3cc8c)

Reply:
Yup, that makes sense to me!

Reply:
Cool, also added a bullet and the mock for this in the categories section of the high level for posterity.

Reply:
I think this is why it would be nice to have the wrapper component so it can display this custom empty state too

Reply:
Could you share the Figma link for this component? I need access to the assets. Thanks!

Reply:
Thanks @trjExpensify! I've updated empty state component